libraries/AP_Param/AP_Param.cpp

```diff
@@ -1789,9 +1803,20 @@ bool AP_Param::load_defaults_file(const char *filename)
         return false;
     }
 
-    if (! read_param_defaults_file(filename)) {
-        return false;
+    saveptr = nullptr;
+    mutable_filename = strdup(filename);
+    if (mutable_filename == nullptr) {
+        AP_HAL::panic("AP_Param: Failed to allocate mutable string");
+    }
+    for (char *pname = strtok_r(mutable_filename, ",", &saveptr);
+         pname != nullptr;
+         pname = strtok_r(nullptr, ",", &saveptr)) {
+        if (!read_param_defaults_file(pname)) {
+            free(mutable_filename);
+            return false;
+        }
```

Comment:
why are you doing this twice in the same function instead of just reusing the modified string?